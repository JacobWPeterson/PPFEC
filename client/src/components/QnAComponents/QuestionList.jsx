import React from 'react';
import Item from './Item.jsx';

const QuestionList = (props) => (
  <div>
    {props.items.map((item) => (<Item item={item} key={item.question_id} id={item.question_id} />))}
  </div>

);

export default QuestionList;
