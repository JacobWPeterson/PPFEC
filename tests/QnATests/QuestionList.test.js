import React from 'react';
import { mount } from 'enzyme';
import QuestionList from '../../client/src/components/QnAComponents/QuestionList.jsx';

describe('This is for the Question List', () => {
  const props = {
    items: [
      {
        answers: {
          768676: {
            answerer_name: 'Koby72',
            body: 'Laborum voluptatum aut et ex occaecati magni quas.',
            date: '2020-07-28T00:00:00.000Z',
            helpfulness: 15,
            id: 768676,
            photos: [],
          },
          768699: {
            answerer_name: 'Ko2',
            body: 'Laborum volupt aut et ex occaecati magni quas.',
            date: '2020-07-28T00:00:00.000Z',
            helpfulness: 15,
            id: 768699,
            photos: [],
          },
        },
        asker_name: 'Maxine.Kilback16',
        question_body: 'Fugit in recusandae delectus ut.',
        question_date: '2020-03-19T00:00:00.000Z',
        question_helpfulness: 30,
        question_id: 80975,
        reported: false,
      },
      {
        answers: {
          768676: {
            answerer_name: 'Koby72',
            body: 'Laborum voluptatum aut et ex occaecati magni quas.',
            date: '2020-07-28T00:00:00.000Z',
            helpfulness: 15,
            id: 768676,
            photos: [],
          },
          768699: {
            answerer_name: 'Ko2',
            body: 'Laborum volupt aut et ex occaecati magni quas.',
            date: '2020-07-28T00:00:00.000Z',
            helpfulness: 15,
            id: 768699,
            photos: [],
          },
        },
        asker_name: 'Maxine.Kilback16',
        question_body: 'Fugit in recusandae delectus ut.',
        question_date: '2020-03-19T00:00:00.000Z',
        question_helpfulness: 30,
        question_id: 80100,
        reported: false,
      },
    ],
    name: 'This is a product',
  };

  const questionListComp = mount(<QuestionList {...props} />);

  it('Should render More Answers Properly', () => {
    expect(questionListComp).toMatchSnapshot();
  });
});
