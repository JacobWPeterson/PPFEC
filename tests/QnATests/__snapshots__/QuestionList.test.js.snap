// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the Question List Should render More Answers Properly 1`] = `
<QuestionList
  items={
    Array [
      Object {
        "answers": Object {
          "768676": Object {
            "answerer_name": "Koby72",
            "body": "Laborum voluptatum aut et ex occaecati magni quas.",
            "date": "2020-07-28T00:00:00.000Z",
            "helpfulness": 15,
            "id": 768676,
            "photos": Array [],
          },
          "768699": Object {
            "answerer_name": "Ko2",
            "body": "Laborum volupt aut et ex occaecati magni quas.",
            "date": "2020-07-28T00:00:00.000Z",
            "helpfulness": 15,
            "id": 768699,
            "photos": Array [],
          },
        },
        "asker_name": "Maxine.Kilback16",
        "question_body": "Fugit in recusandae delectus ut.",
        "question_date": "2020-03-19T00:00:00.000Z",
        "question_helpfulness": 30,
        "question_id": 80975,
        "reported": false,
      },
      Object {
        "answers": Object {
          "768676": Object {
            "answerer_name": "Koby72",
            "body": "Laborum voluptatum aut et ex occaecati magni quas.",
            "date": "2020-07-28T00:00:00.000Z",
            "helpfulness": 15,
            "id": 768676,
            "photos": Array [],
          },
          "768699": Object {
            "answerer_name": "Ko2",
            "body": "Laborum volupt aut et ex occaecati magni quas.",
            "date": "2020-07-28T00:00:00.000Z",
            "helpfulness": 15,
            "id": 768699,
            "photos": Array [],
          },
        },
        "asker_name": "Maxine.Kilback16",
        "question_body": "Fugit in recusandae delectus ut.",
        "question_date": "2020-03-19T00:00:00.000Z",
        "question_helpfulness": 30,
        "question_id": 80100,
        "reported": false,
      },
    ]
  }
  name="This is a product"
>
  <QuestionList__QList>
    <div
      className="QuestionList__QList-ar56kx-0 eOppjS"
    >
      <Item
        id={80975}
        item={
          Object {
            "answers": Object {
              "768676": Object {
                "answerer_name": "Koby72",
                "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                "date": "2020-07-28T00:00:00.000Z",
                "helpfulness": 15,
                "id": 768676,
                "photos": Array [],
              },
              "768699": Object {
                "answerer_name": "Ko2",
                "body": "Laborum volupt aut et ex occaecati magni quas.",
                "date": "2020-07-28T00:00:00.000Z",
                "helpfulness": 15,
                "id": 768699,
                "photos": Array [],
              },
            },
            "asker_name": "Maxine.Kilback16",
            "question_body": "Fugit in recusandae delectus ut.",
            "question_date": "2020-03-19T00:00:00.000Z",
            "question_helpfulness": 30,
            "question_id": 80975,
            "reported": false,
          }
        }
        key="80975"
        name="This is a product"
      >
        <div>
          <Item__TopWrapper>
            <div
              className="Item__TopWrapper-nd8d2l-0 dafvsE"
            >
              <Item__QB>
                <h4
                  className="Item__QB-nd8d2l-8 fxalhT"
                >
                  <Item__Q>
                    <div
                      className="Item__Q-nd8d2l-7 xnwUm"
                    >
                      Q:
                    </div>
                  </Item__Q>
                  Fugit in recusandae delectus ut.
                </h4>
              </Item__QB>
              <Item__Helper>
                <div
                  className="Item__Helper-nd8d2l-2 cFkmUj"
                >
                  <div>
                    Helpful?
                  </div>
                  <Item__Yes
                    onClick={[Function]}
                  >
                    <div
                      className="Item__Yes-nd8d2l-3 czRgbS"
                      onClick={[Function]}
                    >
                      Yes
                    </div>
                  </Item__Yes>
                  <Item__Helpfulness>
                    <div
                      className="Item__Helpfulness-nd8d2l-5 jZuxNN"
                    >
                      (30)
                    </div>
                  </Item__Helpfulness>
                  <div>
                    |
                  </div>
                  <Item__AddAButton
                    onClick={[Function]}
                  >
                    <div
                      className="Item__AddAButton-nd8d2l-4 jOspkh"
                      onClick={[Function]}
                    >
                      Add Answer
                    </div>
                  </Item__AddAButton>
                </div>
              </Item__Helper>
            </div>
          </Item__TopWrapper>
          <Item__BottomWrapper>
            <div
              className="Item__BottomWrapper-nd8d2l-1 iZhiUt"
            >
              <Item__A>
                <div
                  className="Item__A-nd8d2l-6 kwDvjH"
                >
                  A:
                </div>
              </Item__A>
              <Item__ListContainer>
                <div
                  className="Item__ListContainer-nd8d2l-9 GlEbY"
                >
                  <AnswerList
                    answers={
                      Array [
                        Object {
                          "answerer_name": "Koby72",
                          "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                          "date": "2020-07-28T00:00:00.000Z",
                          "helpfulness": 15,
                          "id": 768676,
                          "photos": Array [],
                        },
                        Object {
                          "answerer_name": "Ko2",
                          "body": "Laborum volupt aut et ex occaecati magni quas.",
                          "date": "2020-07-28T00:00:00.000Z",
                          "helpfulness": 15,
                          "id": 768699,
                          "photos": Array [],
                        },
                      ]
                    }
                  >
                    <div>
                      <Answer
                        answer={
                          Object {
                            "answerer_name": "Koby72",
                            "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                            "date": "2020-07-28T00:00:00.000Z",
                            "helpfulness": 15,
                            "id": 768676,
                            "photos": Array [],
                          }
                        }
                        id={768676}
                        key="768676"
                      >
                        <div>
                          <Answer__Body>
                            <div
                              className="Answer__Body-sc-3mvxjy-6 gnmbP"
                            >
                              Laborum voluptatum aut et ex occaecati magni quas.
                            </div>
                          </Answer__Body>
                          <div>
                            <Photos
                              photos={Array []}
                            >
                              <div />
                            </Photos>
                          </div>
                          <Answer__BottomContainer>
                            <div
                              className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                            >
                              <Answer__NameDate>
                                <div
                                  className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                >
                                  By Koby72 on  July, 28 2020
                                </div>
                              </Answer__NameDate>
                              |
                              <Answer__Helpful>
                                <div
                                  className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                >
                                  Helpful?
                                </div>
                              </Answer__Helpful>
                              <Answer__Yes
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                  onClick={[Function]}
                                >
                                  Yes
                                </div>
                              </Answer__Yes>
                              <Answer__Help>
                                <div
                                  className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                >
                                  (15)
                                </div>
                              </Answer__Help>
                              |
                              <Answer__Report
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                  onClick={[Function]}
                                >
                                  Report
                                </div>
                              </Answer__Report>
                            </div>
                          </Answer__BottomContainer>
                        </div>
                      </Answer>
                      <Answer
                        answer={
                          Object {
                            "answerer_name": "Ko2",
                            "body": "Laborum volupt aut et ex occaecati magni quas.",
                            "date": "2020-07-28T00:00:00.000Z",
                            "helpfulness": 15,
                            "id": 768699,
                            "photos": Array [],
                          }
                        }
                        id={768699}
                        key="768699"
                      >
                        <div>
                          <Answer__Body>
                            <div
                              className="Answer__Body-sc-3mvxjy-6 gnmbP"
                            >
                              Laborum volupt aut et ex occaecati magni quas.
                            </div>
                          </Answer__Body>
                          <div>
                            <Photos
                              photos={Array []}
                            >
                              <div />
                            </Photos>
                          </div>
                          <Answer__BottomContainer>
                            <div
                              className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                            >
                              <Answer__NameDate>
                                <div
                                  className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                >
                                  By Ko2 on  July, 28 2020
                                </div>
                              </Answer__NameDate>
                              |
                              <Answer__Helpful>
                                <div
                                  className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                >
                                  Helpful?
                                </div>
                              </Answer__Helpful>
                              <Answer__Yes
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                  onClick={[Function]}
                                >
                                  Yes
                                </div>
                              </Answer__Yes>
                              <Answer__Help>
                                <div
                                  className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                >
                                  (15)
                                </div>
                              </Answer__Help>
                              |
                              <Answer__Report
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                  onClick={[Function]}
                                >
                                  Report
                                </div>
                              </Answer__Report>
                            </div>
                          </Answer__BottomContainer>
                        </div>
                      </Answer>
                    </div>
                  </AnswerList>
                  <MoreA
                    display={false}
                    func={[Function]}
                  >
                    <div />
                  </MoreA>
                </div>
              </Item__ListContainer>
            </div>
          </Item__BottomWrapper>
        </div>
      </Item>
      <Item
        id={80100}
        item={
          Object {
            "answers": Object {
              "768676": Object {
                "answerer_name": "Koby72",
                "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                "date": "2020-07-28T00:00:00.000Z",
                "helpfulness": 15,
                "id": 768676,
                "photos": Array [],
              },
              "768699": Object {
                "answerer_name": "Ko2",
                "body": "Laborum volupt aut et ex occaecati magni quas.",
                "date": "2020-07-28T00:00:00.000Z",
                "helpfulness": 15,
                "id": 768699,
                "photos": Array [],
              },
            },
            "asker_name": "Maxine.Kilback16",
            "question_body": "Fugit in recusandae delectus ut.",
            "question_date": "2020-03-19T00:00:00.000Z",
            "question_helpfulness": 30,
            "question_id": 80100,
            "reported": false,
          }
        }
        key="80100"
        name="This is a product"
      >
        <div>
          <Item__TopWrapper>
            <div
              className="Item__TopWrapper-nd8d2l-0 dafvsE"
            >
              <Item__QB>
                <h4
                  className="Item__QB-nd8d2l-8 fxalhT"
                >
                  <Item__Q>
                    <div
                      className="Item__Q-nd8d2l-7 xnwUm"
                    >
                      Q:
                    </div>
                  </Item__Q>
                  Fugit in recusandae delectus ut.
                </h4>
              </Item__QB>
              <Item__Helper>
                <div
                  className="Item__Helper-nd8d2l-2 cFkmUj"
                >
                  <div>
                    Helpful?
                  </div>
                  <Item__Yes
                    onClick={[Function]}
                  >
                    <div
                      className="Item__Yes-nd8d2l-3 czRgbS"
                      onClick={[Function]}
                    >
                      Yes
                    </div>
                  </Item__Yes>
                  <Item__Helpfulness>
                    <div
                      className="Item__Helpfulness-nd8d2l-5 jZuxNN"
                    >
                      (30)
                    </div>
                  </Item__Helpfulness>
                  <div>
                    |
                  </div>
                  <Item__AddAButton
                    onClick={[Function]}
                  >
                    <div
                      className="Item__AddAButton-nd8d2l-4 jOspkh"
                      onClick={[Function]}
                    >
                      Add Answer
                    </div>
                  </Item__AddAButton>
                </div>
              </Item__Helper>
            </div>
          </Item__TopWrapper>
          <Item__BottomWrapper>
            <div
              className="Item__BottomWrapper-nd8d2l-1 iZhiUt"
            >
              <Item__A>
                <div
                  className="Item__A-nd8d2l-6 kwDvjH"
                >
                  A:
                </div>
              </Item__A>
              <Item__ListContainer>
                <div
                  className="Item__ListContainer-nd8d2l-9 GlEbY"
                >
                  <AnswerList
                    answers={
                      Array [
                        Object {
                          "answerer_name": "Koby72",
                          "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                          "date": "2020-07-28T00:00:00.000Z",
                          "helpfulness": 15,
                          "id": 768676,
                          "photos": Array [],
                        },
                        Object {
                          "answerer_name": "Ko2",
                          "body": "Laborum volupt aut et ex occaecati magni quas.",
                          "date": "2020-07-28T00:00:00.000Z",
                          "helpfulness": 15,
                          "id": 768699,
                          "photos": Array [],
                        },
                      ]
                    }
                  >
                    <div>
                      <Answer
                        answer={
                          Object {
                            "answerer_name": "Koby72",
                            "body": "Laborum voluptatum aut et ex occaecati magni quas.",
                            "date": "2020-07-28T00:00:00.000Z",
                            "helpfulness": 15,
                            "id": 768676,
                            "photos": Array [],
                          }
                        }
                        id={768676}
                        key="768676"
                      >
                        <div>
                          <Answer__Body>
                            <div
                              className="Answer__Body-sc-3mvxjy-6 gnmbP"
                            >
                              Laborum voluptatum aut et ex occaecati magni quas.
                            </div>
                          </Answer__Body>
                          <div>
                            <Photos
                              photos={Array []}
                            >
                              <div />
                            </Photos>
                          </div>
                          <Answer__BottomContainer>
                            <div
                              className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                            >
                              <Answer__NameDate>
                                <div
                                  className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                >
                                  By Koby72 on  July, 28 2020
                                </div>
                              </Answer__NameDate>
                              |
                              <Answer__Helpful>
                                <div
                                  className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                >
                                  Helpful?
                                </div>
                              </Answer__Helpful>
                              <Answer__Yes
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                  onClick={[Function]}
                                >
                                  Yes
                                </div>
                              </Answer__Yes>
                              <Answer__Help>
                                <div
                                  className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                >
                                  (15)
                                </div>
                              </Answer__Help>
                              |
                              <Answer__Report
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                  onClick={[Function]}
                                >
                                  Report
                                </div>
                              </Answer__Report>
                            </div>
                          </Answer__BottomContainer>
                        </div>
                      </Answer>
                      <Answer
                        answer={
                          Object {
                            "answerer_name": "Ko2",
                            "body": "Laborum volupt aut et ex occaecati magni quas.",
                            "date": "2020-07-28T00:00:00.000Z",
                            "helpfulness": 15,
                            "id": 768699,
                            "photos": Array [],
                          }
                        }
                        id={768699}
                        key="768699"
                      >
                        <div>
                          <Answer__Body>
                            <div
                              className="Answer__Body-sc-3mvxjy-6 gnmbP"
                            >
                              Laborum volupt aut et ex occaecati magni quas.
                            </div>
                          </Answer__Body>
                          <div>
                            <Photos
                              photos={Array []}
                            >
                              <div />
                            </Photos>
                          </div>
                          <Answer__BottomContainer>
                            <div
                              className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
                            >
                              <Answer__NameDate>
                                <div
                                  className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
                                >
                                  By Ko2 on  July, 28 2020
                                </div>
                              </Answer__NameDate>
                              |
                              <Answer__Helpful>
                                <div
                                  className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
                                >
                                  Helpful?
                                </div>
                              </Answer__Helpful>
                              <Answer__Yes
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Yes-sc-3mvxjy-3 bpatiV"
                                  onClick={[Function]}
                                >
                                  Yes
                                </div>
                              </Answer__Yes>
                              <Answer__Help>
                                <div
                                  className="Answer__Help-sc-3mvxjy-4 gOKkij"
                                >
                                  (15)
                                </div>
                              </Answer__Help>
                              |
                              <Answer__Report
                                onClick={[Function]}
                              >
                                <div
                                  className="Answer__Report-sc-3mvxjy-5 fkJrli"
                                  onClick={[Function]}
                                >
                                  Report
                                </div>
                              </Answer__Report>
                            </div>
                          </Answer__BottomContainer>
                        </div>
                      </Answer>
                    </div>
                  </AnswerList>
                  <MoreA
                    display={false}
                    func={[Function]}
                  >
                    <div />
                  </MoreA>
                </div>
              </Item__ListContainer>
            </div>
          </Item__BottomWrapper>
        </div>
      </Item>
    </div>
  </QuestionList__QList>
</QuestionList>
`;
